// @apps-in-toss/native-modules/src/CommonModule.brick.ts
import { BrickModule, BrickModuleSpec } from 'brick-module';
import type { HapticFeedbackOptions } from './BedrockModule/native-modules/natives/generateHapticFeedback/types';
import type { NetworkStatus } from './BedrockModule/native-modules/natives/getNetworkStatus/types';

/**
 * CommonModule Spec
 * Unified type definition for CommonModule (formerly BedrockModule)
 */
export interface CommonModuleSpec extends BrickModuleSpec {
  readonly moduleName: 'BedrockModule';

  getConstants(): {
    locale: string;
    schemeUri: string;
  };

  closeView: () => void;
  generateHapticFeedback: (options: HapticFeedbackOptions) => Promise<void>;
  share: (message: { message: string }) => void;
  setSecureScreen: (options: { enabled: boolean }) => Promise<{ enabled: boolean }>;
  setScreenAwakeMode: (options: { enabled: boolean }) => Promise<{ enabled: boolean }>;
  getNetworkStatus: () => Promise<NetworkStatus>;
  setIosSwipeGestureEnabled: ({ isEnabled }: { isEnabled: boolean }) => Promise<void>;
}

/**
 * CommonModule Instance
 * Native module instance accessed through BrickModule
 */
export const CommonModule = BrickModule.get<CommonModuleSpec>('CommonModule');

/**
 * Deprecated exports for backward compatibility
 */
/** @deprecated Use CommonModule instead */
export const BedrockModule = CommonModule;
